src/ios/AppDelegate+BlueshiftPlugin.m

```diff
@@ -9,63 +9,276 @@
+            SEL swizzledSelector = @selector(blueshift_swizzled_application:continueUserActivity:restorationHandler:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        } else {
+            SEL originalSelector = @selector(application:continueUserActivity:restorationHandler:);
+            SEL swizzledSelector = @selector(blueshift_swizzled_no_application:continueUserActivity:restorationHandler:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        }
+        
+        if ([uiApplicationDelegate respondsToSelector:@selector(applicationWillEnterForeground:)]) {
+            SEL originalSelector = @selector(applicationWillEnterForeground:);
+            SEL swizzledSelector = @selector(blueshift_swizzled_applicationWillEnterForeground:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        } else {
+            SEL originalSelector = @selector(applicationWillEnterForeground:);
+            SEL swizzledSelector = @selector(blueshift_swizzled_no_applicationWillEnterForeground:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        }
+        
+        if ([uiApplicationDelegate respondsToSelector:@selector(applicationDidEnterBackground:)]) {
+            SEL originalSelector = @selector(applicationDidEnterBackground:);
+            SEL swizzledSelector = @selector(blueshift_swizzled_applicationDidEnterBackground:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        } else {
+            SEL originalSelector = @selector(applicationDidEnterBackground:);
+            SEL swizzledSelector = @selector(blueshift_swizzled_no_applicationDidEnterBackground:);
+            [self swizzleMethodWithClass:class originalSelector:originalSelector andSwizzledSelector:swizzledSelector];
+        }
+    });
+}
+
++ (void)swizzleMethodWithClass:(Class)class originalSelector:(SEL)originalSelector andSwizzledSelector:(SEL)swizzledSelector {
+    Method originalMethod = class_getInstanceMethod(class, originalSelector);
+    Method swizzledMethod = class_getInstanceMethod(class, swizzledSelector);
+    
+    BOOL isSuccess = class_addMethod(class, originalSelector, method_getImplementation(swizzledMethod), method_getTypeEncoding(swizzledMethod));
+    if (isSuccess) {
+        class_replaceMethod(class, swizzledSelector, method_getImplementation(originalMethod), method_getTypeEncoding(originalMethod));
+    } else {
+        method_exchangeImplementations(originalMethod, swizzledMethod);
+    }
+}
+
+#pragma mark - Remote Notification methods
+- (void)blueshift_swizzled_application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
```

Comment:
is this the correct naming convention? mixing up camelcase and snakecase

Reply:
For method swizzling, the above naming convention is followed by everyone.